src/plugins/xerces/util.hpp

```diff
@@ -0,0 +1,83 @@
+/**
+ * @file
+ *
+ * @brief Utility functions used in the xerces plugin
+ * 
+ * Provides unique pointers that act as a bridge between XMLCh and std::string
+ * Provides a general XercesPluginException
+ *
+ * @copyright BSD License (see doc/LICENSE.md or http://www.libelektra.org)
+ */
+
+#ifndef ELEKTRA_PLUGIN_XERCES_UTIL_H
+#define ELEKTRA_PLUGIN_XERCES_UTIL_H
+
+#include <xercesc/util/XMLString.hpp>
+
+template <typename T>
+struct XercesDeleter
+{
+	void operator() (T * ptr) const
+	{
+		ptr->release ();
+	}
+};
+
+template <>
+struct XercesDeleter<XMLCh>
+{
+	void operator() (XMLCh * ptr) const
+	{
+		XERCES_CPP_NAMESPACE::XMLString::release (&ptr);
+	}
+};
+
+template <>
+struct XercesDeleter<char>
+{
+	void operator() (char * ptr) const
+	{
+		XERCES_CPP_NAMESPACE::XMLString::release (&ptr);
+	}
+};
+
+template <typename T>
+using xerces_unique_ptr = std::unique_ptr<T, XercesDeleter<T>>;
```

Comment:
Yes, or even XercesPtr is okay! There are plenty of such definitions in libtool, e.g. PluginPtr.